src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -81,36 +83,49 @@ public SourceInfo<string> GetConfigMonikerRange(FilePath file)
             return (null, monikers);
         }
 
-        private (Error? error, string[] monikers) GetFileLevelMonikersCore(FilePath file)
+        private (List<Error> errors, string[] monikers) GetFileLevelMonikersCore(FilePath file)
         {
+            var errors = new List<Error>();
             var (_, metadata) = _metadataProvider.GetMetadata(file);
 
             var configMonikerRange = GetConfigMonikerRange(file);
-            var configMonikers = _rangeParser.Parse(configMonikerRange);
+            var configedMonikers = _rangeParser.Parse(configMonikerRange);
 
-            if (!string.IsNullOrEmpty(metadata.MonikerRange))
+            if (!string.IsNullOrEmpty(metadata.MonikerRange) || metadata.Monikers.Length > 0)
             {
                 // Moniker range not defined in docfx.yml/docfx.json,
                 // user should not define it in file metadata
-                if (configMonikers.Length == 0)
+                if (configedMonikers.Length == 0)
                 {
-                    return (Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange), configMonikers);
+                    errors.Add(Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange));
+                    return (errors, configedMonikers);
                 }
 
                 var fileMonikers = _rangeParser.Parse(metadata.MonikerRange);
-                var intersection = configMonikers.Intersect(fileMonikers).ToArray();
+
+                // monikerRagen takes precedence over monikers since it is more likely from user configuration
+                if (fileMonikers.Length > 0 && metadata.Monikers.Length > 0)
+                {
+                    errors.Add(Errors.Versioning.DuplciateMonikerConfig(new SourceInfo(file, 0, 0)));
+                }
+                if (fileMonikers.Length == 0 && metadata.Monikers.Length > 0)
+                {
+                    fileMonikers = metadata.Monikers.Select(x => x.Value).ToArray();
```

Comment:
Now that `monikers` is an input, need to:
- Check that `monikers` are all valid like `monikerRange` (exist in moniker definition)
- Lowercase `monikers` like `monikerRange` since it is case insensitive

Reply:
Check the last test from this PR, error will be thrown if `monikers` out range

Reply:
I mean `monikers` can be `[asdf, netcore-1.0]` where `asdf` is an undefined moniker

Reply:
get your point now, that means `Monikers` should be parsed by `MonikerRangeParser`

Reply:
Oh, I'm not saying the should go thought `MonikerRangeParser`, it just needs the same validation and transform (toLower), a simple method is enough to handle this without _parsing_.

Reply:
why not just use the parser then? it doesn't contain unnecessary logic